components/selectors/matching.rs

```diff
@@ -651,7 +761,12 @@ fn matches_simple_selector<E, F>(
             matches_generic_nth_child(element, 0, 1, true, true, flags_setter)
         }
         Component::Negation(ref negated) => {
-            !negated.iter().all(|ss| matches_simple_selector(ss, element, context, relevant_link, flags_setter))
+            let old_value = context.within_functional_pseudo_class_argument;
+            context.within_functional_pseudo_class_argument = true;
+            let result = !negated.iter().all(|ss| matches_simple_selector(ss, element, context,
```

Comment:
nit: Please add braces and move the function call to its own line, or indent the arguments (though preferably the first).